modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/config/ConfigurationInstanceProviderFactoryBean.java

```diff
@@ -89,4 +85,30 @@ public boolean isSingleton()
     {
         return true;
     }
+
+    private DynamicConfigPolicy getDynamicConfigPolicy(ElementDescriptor element)
+    {
+        ElementDescriptor policyElement = element.getChildByName("dynamic-config-policy");
+        if (policyElement == null)
+        {
+            return NullDynamicConfigPolicy.INSTANCE;
+        }
+
+        return new ImmutableDynamicConfigPolicy(getExpirationPolicy(policyElement));
+    }
+
+    private ExpirationPolicy getExpirationPolicy(ElementDescriptor dynamicConfigPolicyElement)
+    {
+        ElementDescriptor expirationPolicyElement = dynamicConfigPolicyElement.getChildByName("expiration-policy");
+        if (expirationPolicyElement == null)
+        {
+            return new ImmutableExpirationPolicy(5, TimeUnit.MINUTES, TimeSupplier.INSTANCE);
```

Comment:
There are unit tests in which ImmutableExpirationPolicy is used with a TestTimeSupplier.

For the real production code, we use the real one and test it with a FunctionalTestCase